### PR TITLE
code-server: 4.12.0 -> 4.13.0

### PR DESCRIPTION
**Pull request by drupol**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
@GrahamcOfBorg build code-server

**Comment by drupol** [comment]
@GrahamcOfBorg build code-server

**Review by henkery — APPROVED** [review]

**Comment by henkery** [comment]
LGTM :+1:

**Comment by drupol** [comment]
I don't think it's good, it's not building

File: pkgs/servers/code-server/default.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
 { lib, stdenv, fetchFromGitHub, buildGoModule, makeWrapper
 , cacert, moreutils, jq, git, rsync, pkg-config, yarn, python3
-, esbuild, nodejs_16, node-gyp, libsecret, xorg, ripgrep
+, esbuild, nodejs_18, node-gyp, libsecret, xorg, ripgrep
```
Comment (henkery) [inline pkgs/servers/code-server/default.nix:3]: why did you change to nodejs 18, code-server officially still only supports 16
https://github.com/coder/code-server/blob/2109d1cf6abdf17e620a926239d9e96a07face4d/.github/workflows/release.yaml#LL38C13-L38C13

Comment (drupol) [reply pkgs/servers/code-server/default.nix:3]: `nodejs_16` has been deprecated, so I'm just checking if this is working with NodeJS 18. I already made the change in openvscode-server at https://github.com/NixOS/nixpkgs/pull/232887

**Comment by drupol** [comment]
@GrahamcOfBorg build code-server

**Comment by drupol** [comment]
I guess I can close this PR now, it just doesn't work with NodeJS 18.

**Comment by henkery** [comment]
made an issue at code-server to see if they can look into migrating first
https://github.com/coder/code-server/issues/6230

**Comment by drupol** [comment]
Work is continued in https://github.com/NixOS/nixpkgs/pull/234289
